refactor: sf plugin style [HUT-1374]

amtrack opened the pull request:
BREAKING CHANGE: This plugin has been heavily refactored.

- update dependencies and devDependencies and regenerate yarn.lock
- get rid of the keytar dependency
- switch unit testing and stub libraries to what is used in e.g. plugin-deploy-retrieve
- migrate to async/await

amtrack committed:
refactor: sf plugin style

BREAKING CHANGE: This plugin has been heavily refactored.

- update dependencies and devDependencies and regenerate yarn.lock
- switch unit testing and stub libraries to what is used in e.g. plugin-deploy-retrieve
- migrate to async/await

amtrack committed:
docs: cleanup and refer to sf

amtrack committed:
fix: message when creating a branch

amtrack committed:
fix: error handling for HTTP requests

amtrack committed:
fix: handle error when no default org present

mmoyaferrer commented on test/commands/org/authorize.test.ts:
```diff
@@ -1,133 +1,123 @@
 
 describe('hutte:org:authorize', async () => {
-    Authorize.requiresProject = false;
+  const $$ = new TestContext();
```
From our previous conversation, I am aware Salesforce follows this naming convention, but I again feel it's not right. I would personally not follow it and ensure a more readable tests codebase.

amtrack replied:
OK 😄

mmoyaferrer commented on src/common.ts:
```diff
@@ -4,143 +4,105 @@ import { readFileSync, unlinkSync, writeFileSync } from 'fs';
 import { homedir } from 'os';
```
Just to share, at some point (does not have to happen in this PR) I would extract the methods from this `common` class into purpose-specific modules. These `common` classes tend to be a "rubbish bin" to put all methods that a dev does not know where to put, and can be a nightmare of unrelated code in a future.

amtrack replied:
Yes, I agree!

mmoyaferrer reviewed (dismissed):
Great refactor @amtrack , not only for `sf` upgrade but also for some great technical changes along the way. I left a few minor comments but these are minor, feel free to merge.

amtrack committed:
test: rename variable for better readability

amtrack committed:
feat: deprecate keytar [HUT-1399] (#46)

BREAKING CHANGE: We no longer use the deprecated "keytar" npm package to store the API token.

When using "hutte" next time, you'll get an error and need to log in again using "sf hutte auth login".
You can then delete the "hutte-io" entry from the keychain of your OS.

MacOS

- Open "Keychain Access.app"
- Search for "hutte-io" of kind application password
- Right click and delete

Windows

- Open "Credential Manager"
- In the "Generic Credential" section, search for "hutte-io"
- Click "Remove"
